### ISSUE TITLE
ClassCastException, when use Proxy instead of Session

### ISSUE THREAD

**Issue by AndreevIgor** [arey/hibernate-hydrate#8]
When use Proxy instead of Session I get the following error:
> java.lang.ClassCastException: $Proxy65 cannot be cast to org.hibernate.impl.AbstractSessionImpl

In lines:
com.javaetmoi.core.persistence.hibernate.LazyLoadingUtil.deepInflateEntity()
```
Serializable identifier = classMetadata.getIdentifier(entity, (AbstractSessionImpl) currentSession);
```

But, if I change casting to SessionImplementor, there is no error:
```
Serializable identifier = classMetadata.getIdentifier(entity, (SessionImplementor) currentSession);
```

**Comment by arey** [comment]
Thanks AndreevIgor for reporting this issue. I commit the change and I will soon release a 2.1 version.

**Comment by AndreevIgor** [comment]
Thanks.
Is it possible to make this fix and for Hibernate 3.x version too?

**Comment by arey** [comment]
Yes of course. I have also released the 1.3 version for Hibernate 3.x
